modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/RuleEditor/RuleEditor.es.js

```diff
@@ -916,11 +916,27 @@ class RuleEditor extends Component {
 				this.refs.confirmationModalCondition.emit('hide');
 			}
 
+			let maxPageIndex = 0;
+
+			const visitor = new PagesVisitor(pages);
+
+			visitor.mapFields(
+				(field, fieldIndex, columnIndex, rowIndex, pageIndex) => {
+					const condition = conditions.find(
+						condition =>
+							condition.operands[0].value == field.fieldName
+					);
+
+					maxPageIndex = condition ? pageIndex : maxPageIndex;
+				}
+			);
```

Comment:
Hey @marcosapmf, this same logic seems to be already available, please check [here](https://github.com/natocesarrego/liferay-portal/blob/811aedad2caf03680099f92d30d809801a0f1e84/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/RuleEditor/RuleEditor.es.js#L387). Can you please check it and re-send?

Thanks.